core/unix/src/TUnixSigHandling.cxx

```diff
@@ -0,0 +1,903 @@
+{ }
+
+////////////////////////////////////////////////////////////////////////////////
+/// Reset to original state.
+
+TUnixSigHandling::~TUnixSigHandling()
+{
+   UnixResetSignals();
+   delete fSignals;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Initialize Unix system interface.
+
+void TUnixSigHandling::Init()
+{
+   TSigHandling::Init();
+
+   fSignals    = new TFdSet;
+
+   //--- install default handlers
+   UnixSignal(kSigChild,                 SigHandler);
+   UnixSignal(kSigBus,                   SigHandler);
+   UnixSignal(kSigSegmentationViolation, SigHandler);
+   UnixSignal(kSigIllegalInstruction,    SigHandler);
+   UnixSignal(kSigSystem,                SigHandler);
+   UnixSignal(kSigPipe,                  SigHandler);
+   UnixSignal(kSigAlarm,                 SigHandler);
+   UnixSignal(kSigUrgent,                SigHandler);
+   UnixSignal(kSigFloatingException,     SigHandler);
+   UnixSignal(kSigWindowChanged,         SigHandler);
+
+#if defined(R__MACOSX)
+   // trap loading of all dylibs to register dylib name,
+   // sets also ROOTSYS if built without ROOTPREFIX
+   _dyld_register_func_for_add_image(DylibAdded);
+#elif defined(HAVE_DLADDR)
+   SetRootSys();
+#endif
+
+#ifndef ROOTPREFIX
+   gRootDir = Getenv("ROOTSYS");
+   if (gRootDir == 0)
+      gRootDir= "/usr/local/root";
+#else
+   gRootDir = ROOTPREFIX;
+#endif
+
+   if(snprintf(gStackTraceHelper.fShellExec, kStringLength-1, "/bin/sh") >= kStringLength) {
+      SignalSafeErrWrite("Unable to pre-allocate shell command path");
+      return;
+   }
+
+#ifdef ROOTETCDIR
+   if(snprintf(gStackTraceHelper.fPidString, kStringLength-1, "%s/gdb-backtrace.sh", ROOTETCDIR) >= kStringLength) {
+      SignalSafeErrWrite("Unable to pre-allocate executable information");
+      return;
+   }
+#else
+   if(snprintf(gStackTraceHelper.fPidString, kStringLength-1, "%s/etc/gdb-backtrace.sh", gSystem->Getenv("ROOTSYS")) >= kStringLength) {
+      SignalSafeErrWrite("Unable to pre-allocate executable information");
+      return;
+   }   
+#endif
+
+   gStackTraceHelper.fParentToChild[0] = -1;
+   gStackTraceHelper.fParentToChild[1] = -1;
+   gStackTraceHelper.fChildToParent[0] = -1;
+   gStackTraceHelper.fChildToParent[1] = -1;
+
+   StackTraceHelperInit();
+}
+
+//---- Misc --------------------------------------------------------------------
+
+////////////////////////////////////////////////////////////////////////////////
+/// Get environment variable.
+
+const char *TUnixSigHandling::Getenv(const char *name)
+{
+   return ::getenv(name);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Get process id.
+
+int TUnixSigHandling::GetPid()
+{
+   return ::getpid();
+}
+
+////////////////////////////////////////////////////////////////////////////////
+///// Exit the application.
+
+void TUnixSigHandling::Exit(int code, Bool_t mode)
```

Comment:
Rather than duplication gSystem's function, would it make sense to use gSystem instead of implementing this new function?